sonar/sonar_setup.sh

```
@@ -0,0 +1,125 @@
+# Complete SonarQube Setup and Analysis Script
+# This script handles everything: Docker setup, project creation, token generation, and Gradle analysis
+
+set -e  # Exit on any error
+
+# Configuration
+SONAR_URL="http://localhost:9000"
+SONAR_USER="YOUR_USER"
+SONAR_PASS="YOUR_PASSWORD"
+PROJECT_KEY="${1:-YOUR_PROJECT}"  # Use command line argument or default
+BUILD_GRADLE_FILE="build.gradle"
+DOCKER_COMPOSE_FILE="docker-compose.yml"
```

Shellcheck (0.10.0)

[warning] DOCKER_COMPOSE_FILE appears unused. Verify use (or export if used externally).

(SC2034)